Libraries/LibJS/Interpreter.h

```diff
@@ -27,33 +27,45 @@
 #pragma once
 
 #include <AK/Vector.h>
+#include <AK/HashMap.h>
+#include <AK/String.h>
 #include <LibJS/Forward.h>
 
 namespace JS {
 
-struct ScopeFrame {
-    const ScopeNode& scope_node;
+class ScopeFrame {
+public:
+    ScopeFrame(const ScopeNode& scope_node, const ScopeFrame* parent);
+
+    Value get_var(String name) const;
+    void put_var(String name, Value value);
+
+    const ScopeNode& get_scope_node() const { return m_scope_node; }
```

Comment:
Please do not use `get_` prefixes, per https://github.com/SerenityOS/serenity/blob/master/Documentation/CodingStyle.md#wrong-3